core/txpool/legacypool/list.go

```diff
@@ -160,13 +160,11 @@ func (m *sortedMap) Cap(threshold int) types.Transactions {
 	// Otherwise gather and drop the highest nonce'd transactions
 	var drops types.Transactions
 
-	sort.Sort(*m.index)
 	for size := len(m.items); size > threshold; size-- {
-		drops = append(drops, m.items[(*m.index)[size-1]])
-		delete(m.items, (*m.index)[size-1])
+		dropIdx := m.index.Pop().(uint64)
```

Comment:
Doens't this negate the previous algo? popping removes from the front of the list, not the back??

Reply:
Yeah. forgive me. this is a mistake. reverted.

Reply:
the sort is unavoidable here.